Dockerfile

```diff
@@ -0,0 +1,40 @@
+ARG BALENA_ARCH=%%BALENA_ARCH%%
+ARG DISTRO=debian
+
+FROM balenalib/${BALENA_ARCH}-debian:buster-build AS debian
```

Comment:
Just a heads up that for armv6hf we were using `balenalib/rpi-raspbian`, but I'm not sure whether we really have to.

Reply:
were we publishing to S3 with `rpi` as the arch in the filename as well?

Reply:
@klutchell it looks like the answer is a bit complicated...
There is no `balenalib/armv6-*`
See: https://hub.docker.com/search?q=balenalib%2Farmv6
The most recent tag for `balenalib/rpi-raspbian` is from 9 months ago, so I guess we moved away from that in favor of `balenalib/rpi-debian`
See: https://hub.docker.com/r/balenalib/rpi-raspbian/tags
See: https://hub.docker.com/r/balenalib/rpi-debian/tags

and fwiw we also have `balenalib/rpi-alpine`, so it looks like that we never used the `armv6hf` slug in favor of `rpi` (which is true for other parts of the stack as well)
See: https://hub.docker.com/r/balenalib/rpi-alpine

Reply:
FWIW in this PR, I think the only think that needs changing is using `rpi` instead of `armv6hf` ONLY for the base image used. The rest seems correct afaict :+1: